### PR TITLE
Bumped avvio to 7.0.0-alpha.1

### PR DESCRIPTION
**Pull request by mcollina**
Updated, and it still shows some bugs in Avvio.

### CODE REVIEW

File: test/register.test.js [hunk 1]
```diff
@@ -128,31 +128,23 @@ test('awaitable register and after', async t => {
...
-  // TODO this is a bug, it should be false
-  t.is(second, true)
+  await t.rejects(fastify.after(), e)
 
   fastify.register(async (instance, opts, next) => {
-    third = true
+    t.fail('should not be executed')
   })
 
   await t.rejects(fastify.ready())
```
Comment (Eomm) [inline test/register.test.js]:
```suggestion
  await t.rejects(() => fastify.ready(), Error('kaboom'))
```

File: test/register.test.js [hunk 2]
```diff
@@ -128,31 +128,23 @@ test('awaitable register and after', async t => {
...
+  // TODO this should throw for some reason, and it's not.
+  await fastify.register(async (instance, opts) => {
+    throw e
   })
-  t.is(first, true)
 
-  fastify.register(async (instance, opts, next) => {
-    second = true
+  fastify.register(async (instance, opts) => {
+    t.fail('should not be executed')
```
Comment (Eomm) [inline test/register.test.js]: This is executed: the error is bumped to the `ready` but it doesn't stop the loading (adding chaos to an application startup)

File: test/register.test.js [hunk 3]
```diff
@@ -128,31 +128,23 @@ test('awaitable register and after', async t => {
...
   })
 
-  await fastify.after()
-  // TODO this is a bug, it should be false
-  t.is(second, true)
+  await t.rejects(fastify.after(), e)
```
Comment (Eomm) [inline test/register.test.js]: after should not throw https://github.com/mcollina/avvio/blob/fba94f7cd1572cf400d38bcd64a7c37bf2a46f46/test/await-after.test.js#L152

```suggestion
  await fastify.after()
```

File: test/register.test.js (hunk 2 above)
Comment (delvedor) [reply test/register.test.js]: The test are passing, how can you tell that this is executed?

Comment (Eomm) [reply test/register.test.js]: I tried locally and it was not passing (I run `npm update` without cleaning the node module)

Comment (mcollina) [reply test/register.test.js]: you need a npm install, it’s a fixed dependency update

**Comment by mcollina** [comment]
@Eomm this should be good to land now. Please check

**Pull request merged**